osu.Game.Rulesets.Taiko/Skinning/TaikoLegacySkinTransformer.cs

```diff
@@ -81,11 +81,38 @@ public Drawable GetDrawableComponent(ISkinComponent component)
                         return new LegacyBarLine();
 
                     return null;
+
+                case TaikoSkinComponents.TaikoExplosionGood:
+                case TaikoSkinComponents.TaikoExplosionGreat:
+                case TaikoSkinComponents.TaikoExplosionMiss:
+
+                    var sprite = this.GetAnimation(getHitname(taikoComponent.Component), true, false);
+                    if (sprite != null)
+                        return new LegacyHitExplosion(sprite);
+
+                    return null;
             }
 
             return source.GetDrawableComponent(component);
         }
 
+        private string getHitname(TaikoSkinComponents component)
+        {
+            switch (component)
+            {
+                case TaikoSkinComponents.TaikoExplosionMiss:
+                    return "taiko-hit0";
+
+                case TaikoSkinComponents.TaikoExplosionGood:
+                    return "taiko-hit100";
+
+                case TaikoSkinComponents.TaikoExplosionGreat:
+                    return "taiko-hit300";
+            }
+
+            return string.Empty;
```

Comment:
Out of curiosity, is there a reason this doesn't throw the same way `HitExplosion` does?

Reply:
actually meant to change it here, but still not 100% on this whole solution in the first place.